lib/features/login/data/local/account_cache_manager.dart

```diff
@@ -10,24 +11,29 @@ class AccountCacheManager {
 
   AccountCacheManager(this._accountCacheClient);
 
-  Future<PersonalAccount> getSelectedAccount() async {
-    try {
-      final allAccounts = await _accountCacheClient.getAll();
-      return allAccounts.firstWhere((account) => account.isSelected)
-          .toAccount();
-    } catch (e) {
-      logError('AccountCacheManager::getSelectedAccount(): $e');
+  Future<PersonalAccount> getCurrentAccount() async {
+    final allAccounts = await _accountCacheClient.getAll();
+    log('AccountCacheManager::getCurrentAccount::allAccounts(): $allAccounts');
+    final accountCache = allAccounts.firstWhereOrNull((account) => account.isSelected);
+    log('AccountCacheManager::getCurrentAccount::accountCache(): $accountCache');
+    if (accountCache != null) {
+      return accountCache.toAccount();
+    } else {
       throw NotFoundAuthenticatedAccountException();
     }
   }
 
-  Future<void> setSelectedAccount(PersonalAccount account) {
-    log('AccountCacheManager::setSelectedAccount(): $_accountCacheClient');
+  Future<void> setCurrentAccount(PersonalAccount account) async {
+    log('AccountCacheManager::setCurrentAccount(): $account');
+    final accountCacheExist = await _accountCacheClient.isExistTable();
+    if (accountCacheExist) {
+      await _accountCacheClient.clearAllData();
```

Comment:
- Move it to migration script.
- set all other account to false in `isSelected`
- delete all duplicated account ()